sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -345,15 +346,18 @@ object PartitioningUtils {
    */
   def resolvePartitions(
       pathsWithPartitionValues: Seq[(Path, PartitionValues)],
+      caseSensitive: Boolean,
       timeZone: TimeZone): Seq[PartitionValues] = {
     if (pathsWithPartitionValues.isEmpty) {
       Seq.empty
     } else {
-      // TODO: Selective case sensitivity.
-      val distinctPartColNames =
-        pathsWithPartitionValues.map(_._2.columnNames.map(_.toLowerCase())).distinct
+      val distinctPartColNames = if (caseSensitive) {
+        pathsWithPartitionValues.map(_._2.columnNames)
+      } else {
+        pathsWithPartitionValues.map(_._2.columnNames.map(_.toLowerCase()))
+      }
       assert(
-        distinctPartColNames.size == 1,
+        distinctPartColNames.distinct.size == 1,
         listConflictingPartitionColumns(pathsWithPartitionValues))
```

Comment:
why don't we use `distinctPartColNames` as parameter here? Moreover, is that method working fine according to case sensitivity?

Reply:
The method `listConflictingPartitionColumns` also shows the suspicious paths.
If case sensitive, the method works fine. 
If case insensitive, it will list all column names without any transformation. e.g. 
```
	Partition column name list #0: a
        Partition column name list #1: A
	Partition column name list #2: B
```
I can fix the method listConflictingPartitionColumns. But seems a bit trivial, we will have to display the original column names instead of  transforming all to lower case .

Reply:
yes I see, thanks for the kind explanation.